### DISCUSSION TITLE
Handling changes to randomisable mods

### DISCUSSION THREAD

**Discussion by Pasi4K5** [ppy/osu#16048] [Q&A]
This topic has already been brought up by @mk56-spn in https://github.com/ppy/osu/pull/13688#issuecomment-877807906 and @bdach suggested to discuss this separately (https://github.com/ppy/osu/pull/13688#issuecomment-877808959), so here we are now.

I definitely see the random mod as something that can (and, to some extent, should) be improved over time to increase the playability of the maps it produces. The only way I can think of to make this possible without breaking all replays is to store the entire beatmap every time a randomisable mod is used (I don't know whether or not this is a possibility).

**Comment by mk56-spn** [comment]
It works with a seed system. All that would have to do is have a sort of value attached to random plays ( gen 1, gen 2 gen3 etc..) so the game knows what RD iteration it belongs to

**Reply by Pasi4K5** [reply]
What does the seed have to do with this? It's not about the seed, it's about the algorithm itself. But about the gen1-gen2-thing, I guess that could work, but you'd need to have all the previous randomisation algorithms implemented in the client.

**Reply by peppy** [reply]
It would just use the version or date of the score.

**Reply by Pasi4K5** [reply]
You mean it would use the date of the score to determine which random mod "version" it has to use when playing the replay?

**Reply by peppy** [reply]
Correct. Scores already have a `build_id` denoting which release build they were used, we would either use that or something similar.

**Reply by Pasi4K5** [reply]
Should I create an issue in this regard?

**Reply by bdach** [reply]
we will do so if necessary. for now this is such a low priority item that it requires no further action for now in my view.
